src/main.rs

```diff
@@ -80,20 +80,20 @@ fn optimize_maze(
         cache.insert(current_maze_hash);
 
         let current_length = find_shortest_path(current_maze, entry, exit, touchpoints)
-            .unwrap_or(usize::MIN);
+            .unwrap_or(usize::MIN); // TODO: Surely, usize::MAX, as in "infinite"? (usize::MIN can also be written as '0'.)
```

Comment:
If current_length = 0, unreachable. We always take the biggest/longest length as our current best. So the worst maze would have a path of 0.